### ISSUE TITLE
Subsequent sdist runs are reusing an old, cached wheel

### ISSUE THREAD

**Issue by brechtm** [cjolowicz/nox-poetry#377]
Typically during development, your package's version doesn't change and pip will install a cached wheel built from an older sdist. For example:

```
nox > python -m pip install --constraint=.nox/unit-3-8-sdist/tmp/requirements.txt file:///Users/brechtm/Documents/Code/rinohtype/dist/rinohtype-0.5.3.tar.gz#egg=rinohtype
Collecting rinohtype
  Using cached rinohtype-0.5.3-py3-none-any.whl
```

A solution could be to pass `--no-cache-dir` to pip when installing the package, though that [may cause issues with older pip versions](https://github.com/pypa/pip/issues/5749).

**Comment by brechtm** [comment]
Passing `--no-cache-dir` to pip works, but will also re-download all dependencies from PyPI which is unnecessary.

**Comment by brechtm** [comment]
I actually ran into this issue before. I believe this is really a bug in pip, but [not one that they are willing to fix](https://github.com/pypa/pip/issues/4183).

Perhaps a better workaround is to do `pip cache remove <package-name>` before the install:

    session.run("pip", "cache", "--verbose", "remove", "rinohtype")

**Comment by cjolowicz** [comment]
We use `pip uninstall` to ensure pip installs the local package. Do you have a reproducible example for me?

**Comment by brechtm** [comment]
When installing from an sdist, pip will first build a wheel from it. That wheel is cached, and reused when installing from an sdist with the same name/version. When you're doing development, the package version typically doesn't change, so you end up testing against an old version of your source code.

Hence, to reproduce:
- Make Nox (with nox-poetry) run tests using the sdist
- Introduce a breaking change in your source code
- Run tests again, which will succeed despite the breaking change

**Comment by cjolowicz** [comment]
Fixed in #523
